test/test_unary_ufuncs.py

```diff
@@ -58,8 +58,42 @@
                      -4988429.2, 4988429.2,
                      -1e20, 1e20)
 _float_extremals = (float('inf'), float('-inf'), float('nan'))
+medium_length = 812
+large_size = (1029, 917)
 
 
+def generate_tensors_from_vals(vals, device, dtype, domain):
+    offset = 63
+
+    assert large_size[1] > (medium_length + offset)
+    assert medium_length % 4 == 0
+    assert len(vals) < medium_length
+
+    # Constructs the large tensor containing vals
+    large_tensor = make_tensor(large_size, device=device, dtype=dtype, low=domain[0], high=domain[1])
+
+    # Inserts the vals at an odd place
+    large_tensor[57][offset:offset + len(vals)] = torch.tensor(vals, device=device, dtype=dtype)
+
+    # Takes a medium sized copy of the large tensor containing vals
+    medium_tensor = large_tensor[57][offset:offset + medium_length]
+
+    # Constructs small tensors (4 elements)
+    small_tensors = (t for t in torch.split(medium_tensor, 4))
```

Comment:
I think we can get rid of small_tensors now. The large tensor should CPU test vectorization (including an unvectorized tail due to its odd size), and the scalar tensors should test the unvectorized CPU code path. This change is not directly related to this PR but because we're now creating large_tensor twice it'll be nice to reclaim some perf.